src/models/thread.ts

```diff
@@ -504,8 +504,42 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         throw new Error("Unsupported function on the thread model");
     }
 
+    public getEventReadUpTo(userId: string, ignoreSynthesized?: boolean): string | null {
+        const isCurrentUser = userId === this.client.getUserId();
+        if (isCurrentUser && this.lastReply()) {
+            // If a thread last activity is prior the first read receipt sent in a thread
+            // we want to consider that this thread has been read
+            const beforeFirstThreadedReceipt = (this.lastReply()?.getTs() ?? 0) < this.room.oldestThreadedReceiptTs;
```

Comment:
We know `lastReply()` exists?

```suggestion
            const beforeFirstThreadedReceipt = (this.lastReply().getTs() ?? 0) < this.room.oldestThreadedReceiptTs;
```

Reply:
I thought this wasn't needed, but it actually is. 
TypeScript has no way to know whether `this.lastReply()` will return the same thing when called twice consecutively

Reply:
then it should be `!` ?

```suggestion
            const beforeFirstThreadedReceipt = this.lastReply()!.getTs() < this.room.oldestThreadedReceiptTs;
```

or factor out `const lastReply = this.lastReply()` ?